handlers/meta/whatsapp/templates.go

```diff
@@ -30,18 +31,26 @@ func GetTemplatePayload(templating *courier.Templating) *Template {
 			component = &Component{Type: comp.Type}
 
 			for _, p := range compParams {
-				if p.Type == "image" {
-					component.Params = append(component.Params, &Param{Type: p.Type, Image: &struct {
-						Link string "json:\"link,omitempty\""
-					}{Link: p.Value}})
-				} else if p.Type == "video" {
-					component.Params = append(component.Params, &Param{Type: p.Type, Video: &struct {
-						Link string "json:\"link,omitempty\""
-					}{Link: p.Value}})
-				} else if p.Type == "document" {
-					component.Params = append(component.Params, &Param{Type: p.Type, Document: &struct {
-						Link string "json:\"link,omitempty\""
-					}{Link: p.Value}})
+				if p.Type != "text" {
+					attType, attURL := handlers.SplitAttachment(p.Value)
+					attType = strings.Split(attType, "/")[0]
+					if attType == "application" {
+						attType = "document"
+					}
```

Comment:
I think this code is simpler to read if we let `attType` stay as "application".. because that _is_ is the attachment type in our world, and use `document` in the WhatsApp payload